test/middleware-test.js

```diff
@@ -28,2293 +28,2377 @@ describe('input-validation middleware tests', function () {
-                    done();
-                });
-        });
-        it('bad query param - over limit', function (done) {
-            request(app)
-                .get('/pets')
-                .set('request-id', '1234')
-                .set('api-version', '1.0')
-                .query({ limit: 150, page: 0 })
-                .expect(400, function (err, res) {
-                    if (err) {
-                        throw err;
-                    }
-                    let moreInfoAsJson = JSON.parse(res.body.more_info);
-                    expect(moreInfoAsJson).to.be.instanceof(Array);
-                    expect(res.body.more_info).to.includes('limit');
-                    done();
-                });
-        });
-        it('bad query param - under limit', function (done) {
-            request(app)
-                .get('/pets')
-                .set('request-id', '1234')
-                .set('api-version', '1.0')
-                .query({ limit: 0, page: 0 })
-                .expect(400, function (err, res) {
-                    if (err) {
-                        throw err;
-                    }
-                    let moreInfoAsJson = JSON.parse(res.body.more_info);
-                    expect(moreInfoAsJson).to.be.instanceof(Array);
-                    expect(res.body.more_info).to.includes('limit');
-                    done();
-                });
-        });
-        it('bad path param - wrong format', function (done) {
-            request(app)
-                .get('/pets/12')
-                .set('request-id', '1234')
-                .set('api-version', '1.0')
-                .query({ limit: '50', page: 0 })
-                .expect(400, function (err, res) {
-                    if (err) {
-                        throw err;
-                    }
-                    let moreInfoAsJson = JSON.parse(res.body.more_info);
-                    expect(moreInfoAsJson).to.be.instanceof(Array);
-                    expect(res.body.more_info).to.includes('petId');
-                    done();
-                });
-        });
-        it('bad body - wrong format nested attribute (not parameters)', function (done) {
-            request(app)
-                .put('/pets')
-                .send([{
-                    name: 'name',
-                    tag: 'tag',
-                    test: {
-                        field1: 1234
-                    }
-                }])
-                .expect(400, function (err, res) {
-                    if (err) {
-                        throw err;
-                    }
-                    let moreInfoAsJson = JSON.parse(res.body.more_info);
-                    expect(moreInfoAsJson).to.be.instanceof(Array);
-                    expect(moreInfoAsJson.length).to.equal(2);
-                    expect(res.body.more_info).to.includes('field1');
-                    done();
+    let servers = [{framework: 'koa'}, {framework: 'express'}];
```

Comment:
Please move the koa tests to a separate file, It will make it more easy to maintain